src/test/java/spoon/test/comment/CommentTest.java

```diff
@@ -110,12 +111,37 @@ public void testCombinedPackageInfoComment() {
 
 	@Test
 	public void testJavaDocComment() {
-		String EOL = System.getProperty("line.separator");
+		//the EOL is taken from JavaDocComment.java, which is commited in git with linux \n
+		//that is true on Windows too.
+		String EOL = "\n";
 
 		Factory f = getSpoonFactory();
 		CtClass<?> type = (CtClass<?>) f.Type().get(JavaDocComment.class);
 
 		CtJavaDoc classJavaDoc = (CtJavaDoc) type.getComments().get(0);
+		//contract: test that java doc is printed correctly
+		String str = classJavaDoc.toString();
+		StringTokenizer st = new StringTokenizer(str, "\n\r");
```

Comment:
> n case of java sources checked out from repository it happens quite often

Yes but here you create a tokenizer only for the source code generated by Spoon, so it should not happen.